src/ansys/pytwin/evaluate/tbrom.py

```diff
@@ -0,0 +1,270 @@
+
+        Parameters
+        ----------
+        on_disk: bool
+            Whether the point file is saved on disk (True) or returned in memory (False)
+        output_file_name: str
+            Path where the point file is written if on_disk is True
+        named_selection: str (optional)
+            Named selection on which the point file has to be generated
+        """
+        pointpath = os.path.join(self._tbrom_path, TbRom.OUT_F_KEY, TbRom.TBROM_POINTS)
+        vec = np.array(TbRom._read_binary(pointpath))
+        if namedselection is not None:
+            pointsids = self.namedselectionids(namedselection)
+            listids = []
+            for i in pointsids:
+                for k in range(0, 3):
+                    listids.append(i * 3 + k)
+            vec = vec[listids]
+        if on_disk:
+            TbRom._write_binary(output_file_path, vec)
+            return output_file_path
+        else:
+            return vec
+
+    def snapshot_generation(self, on_disk: bool, output_file_path: str, namedselection: str = None):
+        """
+        Generate a field snapshot based on current states of the TBROM, either in memory or on disk, for the full field
+        or a specific part. If on disk, the snapshot is written in the output_file_name path
+
+        Parameters
+        ----------
+        on_disk: bool
+            Whether the snapshot file is saved on disk (True) or returned in memory (False)
+        output_file_name: str
+            Path where the snapshot file is written if on_disk is True
+        named_selection: str (optional)
+            Named selection on which the snasphot has to be generated
+        """
+        basis = self._outbasis
+        vec = np.zeros(len(basis[0]))
+        nb_mc = len(basis)
+        mc = list(self._outmcs.values())
+        for i in range(nb_mc):
+            mnp = np.array(basis[i])
+            vec = vec + mc[i] * mnp
+        if namedselection is not None:
+            pointsids = self.namedselectionids(namedselection)
+            listids = []
+            for i in pointsids:
+                for k in range(0, self.outputfielddimensionality):
+                    listids.append(i * self.outputfielddimensionality + k)
+            vec = vec[listids]
+        if on_disk:
+            TbRom._write_binary(output_file_path, vec)
+            return output_file_path
+        else:
+            return vec
+
+    def snapshot_projection(self, snapshot: str, fieldname: str = None):
+        """
+        Project a given snapshot file on the basis associated to the input field name 'fieldname'
+
+        Parameters
+        ----------
+        snapshot: str
+            Path of the input field snapshot file
+        fieldname: str (optional)
+            Name of the input field for which the snapshot projection will be performed (it needs to be defined in case
+            the TBROM is parameterized with multiple input fields)
```

Comment:
```suggestion
        snapshot : str
            Path of the snapshot file with the input field.
        fieldname : str, optional
            Name of the input field to project the snapshot on. The name of the field
            must be specified in case the TBROM is parameterized with multiple input fields.
```
I'm not sure I understood this clearly--but I hope you can understand why I was confused and fix any errors I might have made.